### PR TITLE
Preview: implement concurrency to avoid map-pool exhaustion in renderer

### PR DESCRIPTION
**Pull request by dgaubert**

### CODE REVIEW

File: lib/preview/blend.js [hunk 1]
```diff
@@ -16,6 +16,36 @@ module.exports = async function blend ({ coordinates, offsets, dimensions, forma
...
+  }
+
+  return tiles
+}
+
+function getCoordinateBatches ({ coordinates, concurrency }) {
+  const batches = []
+
+  for (let index = 0; index < coordinates.length; index += concurrency) {
+    const from = index
```
Comment (Algunenano) [inline lib/preview/blend.js]: Why not call `index` `from` and avoid renaming it?

File: lib/preview/defaults.js [hunk 2]
```diff
@@ -20,11 +20,24 @@ module.exports = function defaults (options) {
   const quality = options.quality || null
   const limit = options.limit || 19008
   const tileSize = options.tileSize || 256
+  const concurrency = options.concurrency || 32
```
Comment (Algunenano) [inline lib/preview/defaults.js]: Can we set this as `poolMaxWaitingClients` by default if no value is passed instead?

Comment (dgaubert) [reply lib/preview/defaults.js]: The only point in common between `renderer` & `preview` is `getTile` method and there is no way to get that value from `getTile` function... :(

Comment (dgaubert) [reply lib/preview/defaults.js]: BTW, I will pass that value in windshaft.

Comment (dgaubert) [reply lib/preview/defaults.js]: There I have both, the renderer and preview.

Comment (Algunenano) [reply lib/preview/defaults.js]: :ok_hand: Then I'd leave a comment there (or in the README) to note that this value is related to poolMaxWaitingClients (something like `//to match poolMaxWaitingClients`).

**Comment by dgaubert** [comment]
> I'd rather have a concurrent queue to deal with this 

I've checked a couple of modules to achieve that, I did not like them so much...

**Pull request merged**